x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/assignment/planning/AbstractPreserveAllocations.java

```diff
@@ -80,10 +80,12 @@ AssignmentPlan mergePreservedAllocations(AssignmentPlan assignmentPlan) {
             for (Node n : nodes) {
                 int allocations = assignmentsByModelNodeIdPair.getOrDefault(Tuple.tuple(m.id(), n.id()), 0);
                 if (m.currentAllocationsByNodeId().containsKey(n.id())) {
-                    allocations += addPreservedAllocations(n, m);
-                    // As the node has all its available memory we need to manually account memory of models with
-                    // current allocations.
-                    mergedPlanBuilder.accountMemory(m, n);
+                    if (mergedPlanBuilder.getRemainingMemory(n) >= m.memoryBytes()) {
```

Comment:
The assignment wouldn't be allocated to the same node it was on before the downsize. It might get allocated to a different node if that other one had space.

The user will be able to find this out by looking at the trained model stats.

The next question is, "Why would a node get smaller?" The most likely scenario is that autoscaling in ESS has replaced a node with a node with the same node ID but smaller hardware. In this case we'd expect that some other node in the cluster has capacity to take the assignments that will no longer fit, otherwise that would imply a bug in autoscaling.

In self managed clusters it's more likely that nodes will be removed entirely rather than downsized. It's _possible_ that a self managed user could downsize nodes in such a way that the cluster no longer had capacity for what they'd configured to run. But then they've only themselves to blame if they're managing the hardware. And they can still find out what effect they've had by looking at the transform stats.